### DISCUSSION TITLE
Memory Leak on Server with Railway

### DISCUSSION THREAD

**Discussion by fxmb** [medusajs/medusa#2755] [Q&A] [answered]
The medusa server has a memory leak on railway, so the server shuts down after a couple of minutes...
Does this have to do with railway or is this a medusa issue? I precisely followed the docs and it works for a couple of minutes. I am not uploading anything to the server

`FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`

**Comment by olivermrbl** [comment, marked as answer]
@fxmb Have you set up Redis? If you are deploying without a production-ready Redis server, we default to a fake instance, that consumes quite a lot of memory. This might be the culprit.

**Comment by fxmb** [comment]
@olivermrbl thanks for pointing this out. We have configured redis exactly according to [the docs here](https://docs.medusajs.com/deployments/server/deploying-on-railway)

I have not specified any prod specific Redis configurations though... At least there were none referred to in the docs:

> In the same project view:
> 
> Click on the New button.
> Choose the Database option.
> Choose Add Redis.
> A new Redis database will be added to the project view in a few seconds. Click on it to open the database sidebar.
> 
> To find the Redis database URL which you’ll need later:
> 
> Click on the Redis card.
> Choose the Connect tab.
> Copy the Redis Connection URL.

Any help here is highly appreciated 🙏

**Reply by fxmb** [reply]
Issue was solved, we did not specify the Redis URL correctly, sorry about that
